### PR TITLE
Export vsp grpc params to start ticket auto buyer 

### PR DESCRIPTION
**Pull request by vctt94**
This PR depends on https://github.com/decred/dcrwallet/pull/1785.

It export vsp host and pubkey into the `RunTicketBuyerRequest`, so we can start it with a new vsp server

### CODE REVIEW

**Comment by jrick** [comment]
I think it would be better to add these changes onto the #1785 pr.  But before that, this change breaks the current functionality when setting a pool voting and fee address.  If either of those are set, that behavior should be preserved.

**Comment by vctt94** [comment]
Closing as it was added at #1785

**Comment by jrick** [comment]
it wasn't. we'll review this separately here.

**Comment by jrick** [comment]
needs a rebase

File: internal/rpc/rpcserver/server.go [hunk 1]
```diff
@@ -2430,10 +2433,28 @@ func (t *ticketbuyerV2Server) RunTicketBuyer(req *pb.RunTicketBuyerRequest, svr
...
+	if req.VspHost != "" || req.VspPubkey != "" {
+		vspHost = req.VspHost
+		vspPubKey = req.VspPubkey
+		if vspPubKey == "" {
+			return status.Errorf(codes.InvalidArgument, "vsp pubkey can not be null")
+		}
+		if vspHost == "" {
+			return status.Errorf(codes.InvalidArgument, "vsp host can not be null")
+		}
+	}
```
Comment (jrick) [inline internal/rpc/rpcserver/server.go:2448]: we should error if the request contains both legacy stakepoold and vspd options.  Both cannot be set.

Comment (vctt94) [reply internal/rpc/rpcserver/server.go:2460]: True. Added a check for that

File: internal/rpc/rpcserver/server.go [hunk 2]
```diff
@@ -2420,13 +2420,21 @@ func (t *ticketbuyerV2Server) RunTicketBuyer(req *pb.RunTicketBuyerRequest, svr
 	var votingAddress dcrutil.Address
 	var err error
 	if req.VotingAddress != "" {
+		if req.VspHost != "" || req.VspPubkey != "" {
+			return status.Errorf(codes.InvalidArgument,
+				"request contains both legacy stakepoold and vspd options.")
+		}
```
Comment (jrick) [inline internal/rpc/rpcserver/server.go:2426]: the voting address can be specified without using a stakepool.  Only the check with the pool address (where stakepool fees are paid) is needed.

File: internal/rpc/rpcserver/server.go [hunk 3]
```diff
@@ -2437,6 +2445,10 @@ func (t *ticketbuyerV2Server) RunTicketBuyer(req *pb.RunTicketBuyerRequest, svr
 	var vspHost string
 	var vspPubKey string
 	if req.VspHost != "" || req.VspPubkey != "" {
+		if req.VotingAddress != "" || req.PoolAddress != "" {
+			return status.Errorf(codes.InvalidArgument,
+				"request contains both legacy stakepoold and vspd options.")
+		}
```
Comment (jrick) [inline internal/rpc/rpcserver/server.go:2451]: if you keep the check above, this one can be removed (we've already determined that both are not set together)

**Pull request merged**